rust_src/src/process.rs

```diff
@@ -231,12 +236,12 @@ pub fn process_status(process: LispObject) -> LispObject {
         return p;
     }
     let p_ref = p.as_process_or_error();
-    if unsafe { pget_raw_status_new(p_ref.as_ptr()) } != 0 {
+    if p_ref.raw_status_new() {
         unsafe { update_status(p_ref.as_ptr()) };
     }
     let mut status = LispObject::from_raw(p_ref.status);
-    if let Some(c) = status.as_cons() {
-        status = c.car();
+    if status.is_cons() {
+        status = status.xcons().car();
     };
```

Comment:
So why is the `xcons` version better? You are still doing the `is_cons` check, possibly twice.